models/cohortdefinition.go

```diff
@@ -67,7 +69,36 @@ func (h CohortDefinition) GetAllCohortDefinitions() ([]*CohortDefinition, error)
 	return cohortDefinition, meta_result.Error
 }
 
-func (h CohortDefinition) GetAllCohortDefinitionsAndStatsOrderBySizeDesc(sourceId int) ([]*CohortDefinitionStats, error) {
+// Returns any "team project" entries that are matched to _each and every one_ of the
+// cohort definition ids found in uniqueCohortDefinitionIdsList.
+func (h CohortDefinition) GetTeamProjectsThatMatchAllCohortDefinitionIds(uniqueCohortDefinitionIdsList []int) ([]string, error) {
+
+	db2 := db.GetAtlasDB().Db
+	var teamProjects []string
+	// Find any roles that are paired to each and every one of the cohort_definition_id values.
+	// Roles that ony match part of the values are filtered out by the having(count) clause:
+	query := db2.Table(db.GetAtlasDB().Schema+".cohort_definition_sec_role").
```

Comment:
Stating for my own understanding/clarity: 
https://github.com/uc-cdis/cohort-middleware/pull/89/files?diff=unified&w=1#diff-8d6e2a5018c8db98f284089e2369dd804849c0488d7166974d49992d28bad403R79
So based on the distinct here in this view, then this example::

```
sec_role_name | cohort_definition_id
--------------------------------------
foo                      | 8675309
foo                      | 8675309
```
should not be possible, right? Therefore theres no way for the count in this Having clause to double count the same cohort_definition_id such that it matches the length of uniqueCohortDefinitionList without every cohort_definition_id being present in the uniqueCohortDefinitionIdsList.

Reply:
The count for above would return 1. The goal of this check is to ensure that in the scenario below:
```
sec_role_name | cohort_definition_id  
--------------------------------------  
foo                      | 8675308  
foo                      | 8675309 
```
and `uniqueCohortDefinitionIdsList=[8675308, 8675310]` 
the `foo` team is NOT returned. The `Where ... in ...` clause will match,
but the `Having` will not in this case.

Reply:
This is a good scenario to add to the unit tests...

Reply:
Test added.